### PR TITLE
Implement course model

### PR DESCRIPTION
**Pull request by miloaisdino**

### CODE REVIEW

File: src/main/java/tahub/contacts/MainApp.java [hunk 1]
```diff
@@ -90,7 +93,7 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
             initialData = new AddressBook();
         }
 
-        return new ModelManager(initialData, userPrefs);
+        return new ModelManager(initialData, userPrefs, new tahub.contacts.model.course.UniqueCourseList());
```
Comment (Zeepheru) [inline src/main/java/tahub/contacts/MainApp.java:96]: Why not just import it and call `UniqueCourseList()`?

File: src/main/java/tahub/contacts/logic/commands/CourseCommand.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a course to the course book. "
+            + "eg course c/CS2103T n/Software Engineering";
+
+    public static final String MESSAGE_SUCCESS = "New course added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This course already exists in the course book";
+
+    private final Course toAdd;
+
+    /**
+     * Creates an CourseCommand to add the specified {@code Course}
```
Comment (Zeepheru) [inline src/main/java/tahub/contacts/logic/commands/CourseCommand.java:27]: ... a~~n~~ CourseCommand ...

File: src/main/java/tahub/contacts/logic/parser/CourseCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        String courseCode = argMultimap.getValue(PREFIX_CODE).get();
+        String courseName = argMultimap.getValue(PREFIX_NAME).get();
+
+        if (!Course.isValidCourseCode(courseCode)) {
+            throw new ParseException(COURSE_CODE_MESSAGE_CONSTRAINTS);
+        }
+
+        if (!Course.isValidCourseName(courseName)) {
+            throw new ParseException(Course.COURSE_NAME_MESSAGE_CONSTRAINTS);
+        }
```
Comment (Zeepheru) [inline src/main/java/tahub/contacts/logic/parser/CourseCommandParser.java:41]: `COURSE_CODE_MESSAGE_CONSTRAINTS` and `Course.COURSE_NAME_MESSAGE_CONSTRAINTS`, tho both are part of `Course`

File: src/main/java/tahub/contacts/model/ModelManager.java [hunk 4]
```diff
@@ -75,6 +79,37 @@ public void setAddressBookFilePath(Path addressBookFilePath) {
         userPrefs.setAddressBookFilePath(addressBookFilePath);
     }
```
Comment (Zeepheru) [inline src/main/java/tahub/contacts/model/ModelManager.java:81]: Maybe add a `//=========== Course ==================================================================================` similar to the rest of the file

File: src/test/java/tahub/contacts/storage/JsonSerializableCourseListTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+        JsonSerializableCourseList jsonSerialisedCourseList = new JsonSerializableCourseList(courses);
+        UniqueCourseList courseList = jsonSerialisedCourseList.toModelType();
+
+        assertEquals(courses, courseList);
+    }
+
+    @Test
+    public void toModelType_invalidCourseList_throwsIllegalValueException() {
+        String invalidJson = "{\"courses\": [{\"courseCode\": \"cs234\",\"courseName\": \"lol\"}]}";
```
Comment (Zeepheru) [inline src/test/java/tahub/contacts/storage/JsonSerializableCourseListTest.java:33]: "lol"

**Review by Zeepheru — APPROVED** [review]
LGTM

Just small nits

File: src/main/java/tahub/contacts/logic/parser/CourseCommandParser.java (hunk 3 above)
Comment (miloaisdino) [reply src/main/java/tahub/contacts/logic/parser/CourseCommandParser.java:41]: the error messages are different!

File: src/main/java/tahub/contacts/MainApp.java (hunk 1 above)
Comment (miloaisdino) [reply src/main/java/tahub/contacts/MainApp.java]: fixed

File: src/main/java/tahub/contacts/logic/commands/CourseCommand.java (hunk 2 above)
Comment (miloaisdino) [reply src/main/java/tahub/contacts/logic/commands/CourseCommand.java]: fixed

File: src/main/java/tahub/contacts/model/ModelManager.java (hunk 4 above)
Comment (miloaisdino) [reply src/main/java/tahub/contacts/model/ModelManager.java:81]: fixed

**Pull request merged**